src/lazy.jl

```diff
@@ -256,7 +256,7 @@ end
 # returns a `pos` value that notes the next position where parsing should continue
 # this is essentially the `StructUtils.applyeach` implementation for LazyValues w/ type OBJECT
 function applyobject(keyvalfunc, x::LazyValues)
-    pos = getpos(x)
+    pos::Int = getpos(x)
```

Comment:
These shouldn't be necessary, because the fields are all `::Int`:

https://github.com/JuliaIO/JSON.jl/blob/3f7d15f6d6acb54a1abfa10abd8fcad9509be908/src/lazy.jl#L134

Reply:
Yes, I found this annotation helpful for documenting the type of `pos` and making sure it doesn't change.

Reply:
Yes, I am ok with converting (it might catch some overflow errors as a side benefit).
The convert will happen on all assignments to `pos`, including `pos = (ret isa Int && ret > pos) ? ret : skip(val)` or the assignments to `pos` inside the `@nextbyte` macro.